.travis.yml

```diff
@@ -10,97 +10,96 @@ python:
-    env: DJANGO="Django>=1.6,<1.7"
-  - python: "3.5"
-    env: DJANGO="Django>=1.7,<1.8"
+    - python: "3.5"
+      env: TOX_ENV="django14"
+    - python: "3.5"
+      env: TOX_ENV="django15"
+    - python: "3.5"
+      env: TOX_ENV="django16"
+    - python: "3.5"
+      env: TOX_ENV="django17"
 
-  - python: "3.4"
-    env: DJANGO="Django>=1.4,<1.5"
-  - python: "3.4"
-    env: DJANGO="Django>=1.9,<1.10"
-  - python: "3.4"
-    env: DJANGO="Django>=1.10,<1.11"
-  - python: "3.4"
-    env: DJANGO="Django>=1.11,<1.12"
+    - python: "3.4"
+      env: TOX_ENV="django14"
+    - python: "3.4"
+      env: TOX_ENV="django19"
+    - python: "3.4"
+      env: TOX_ENV="django110"
+    - python: "3.4"
+      env: TOX_ENV="django111"
 
-  - python: "3.3"
-    env: DJANGO="Django>=1.4,<1.5"
-  - python: "3.3"
-    env: DJANGO="Django>=1.9,<1.10"
-  - python: "3.3"
-    env: DJANGO="Django>=1.10,<1.11"
-  - python: "3.3"
-    env: DJANGO="Django>=1.11,<1.12"
-  - python: "3.3"
-    env: DJANGO="https://github.com/django/django/tarball/master"
+    - python: "3.3"
+      env: TOX_ENV="django14"
+    - python: "3.3"
+      env: TOX_ENV="django19"
+    - python: "3.3"
+      env: TOX_ENV="django110"
+    - python: "3.3"
+      env: TOX_ENV="django111"
+    - python: "3.3"
+      env: TOX_ENV="djangodev"
 
-  - python: "3.2"
-    env: DJANGO="Django>=1.4,<1.5"
-  - python: "3.2"
-    env: DJANGO="Django>=1.9,<1.10"
-  - python: "3.2"
-    env: DJANGO="Django>=1.10,<1.11"
-  - python: "3.2"
-    env: DJANGO="Django>=1.11,<1.12"
-  - python: "3.2"
-    env: DJANGO="https://github.com/django/django/tarball/master"
+    - python: "3.2"
+      env: TOX_ENV="django14"
+    - python: "3.2"
+      env: TOX_ENV="django19"
+    - python: "3.2"
+      env: TOX_ENV="django110"
+    - python: "3.2"
+      env: TOX_ENV="django111"
+    - python: "3.2"
+      env: TOX_ENV="djangodev"
+
+    - python: "2.6"
+      env: TOX_ENV="django17"
+    - python: "2.6"
+      env: TOX_ENV="django18"
+    - python: "2.6"
+      env: TOX_ENV="django19"
+    - python: "2.6"
+      env: TOX_ENV="django110"
+    - python: "2.6"
+      env: TOX_ENV="django111"
+    - python: "2.6"
+      env: TOX_ENV="djangodev"
 
-  - python: "2.6"
-    env: DJANGO="Django>=1.7,<1.8"
-  - python: "2.6"
-    env: DJANGO="Django>=1.8,<1.9"
-  - python: "2.6"
-    env: DJANGO="Django>=1.9,<1.10"
-  - python: "2.6"
-    env: DJANGO="Django>=1.10,<1.11"
-  - python: "2.6"
-    env: DJANGO="Django>=1.11,<1.12"
-  - python: "2.6"
-    env: DJANGO="https://github.com/django/django/tarball/master"
   allow_failures:
-    - env: DJANGO="Django>=1.11,<1.12"
-    - env: DJANGO="https://github.com/django/django/tarball/master"
+    - env: TOX_ENV="django111"
+    - env: TOX_ENV="djangodev"
 
 before_install:
-- pip install codecov
-
-install:
-  - pip install -q $DJANGO coverage==3.6
+- pip install codecov tox
 
 script:
-  - coverage run --source=polymorphic runtests.py
+  - tox -e py${TRAVIS_PYTHON_VERSION/./}-$TOX_ENV
```

Comment:
I suspect you may be misunderstanding my intention here.

It would be possible to change this to be simply ` - tox`, but then all the combinations of python and django would run in one travis build. By passing `-e py33-django17` (for example), we can run combination per build, and can make use of Travis features like `allow_failures` and `fast_finish`.